client/packages/invoices/src/Prescriptions/LineEditView/LineEditView.tsx

```diff
@@ -1,31 +1,54 @@
-import React, { useEffect, useMemo } from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import {
   BasicSpinner,
-  ModalMode,
+  InvoiceNodeStatus,
+  isEqual,
   NothingHere,
   RouteBuilder,
   useBreadcrumbs,
+  useConfirmOnLeaving,
+  useDirtyCheck,
+  useNavigate,
   useParams,
 } from '@openmsupply-client/common';
 
 import { ItemRowFragment, ListItems } from '@openmsupply-client/system';
 import { AppRoute } from '@openmsupply-client/config';
 import { PageLayout } from './PageLayout';
-import { usePrescription } from '../api';
+import { usePrescription, usePrescriptionLines } from '../api';
 import { AppBarButtons } from './AppBarButtons';
 import { PrescriptionLineEdit } from './PrescriptionLineEdit';
+import { DraftStockOutLine } from '../../types';
+import { Footer } from './Footer';
+import { NavBar } from './NavBar';
 
 export const PrescriptionLineEditView = () => {
   const { invoiceNumber, itemId } = useParams();
   const { setCustomBreadcrumbs } = useBreadcrumbs();
+  const navigate = useNavigate();
 
   const {
     query: { data, loading: isLoading },
   } = usePrescription();
 
+  const {
+    save: { saveLines, isSavingLines },
+    // delete: { deleteLines },
+  } = usePrescriptionLines(data?.id);
+
+  const newItemId = useRef<string>();
+
+  const { isDirty, setIsDirty } = useDirtyCheck();
+
   const lines =
-    data?.lines.nodes.sort((a, b) => a.item.name.localeCompare(b.item.name)) ??
-    [];
+    data?.lines.nodes.sort((a, b) => a.id.localeCompare(b.id)) ?? [];
+
+  const invoiceId = data?.id ?? '';
+  const status = data?.status;
+
+  const [allDraftLines, setAllDraftLines] = useState<
+    Record<string, DraftStockOutLine[]>
+  >({});
```

Comment:
So this object keeps the draft changes for all the items (indexed by Item ID), and we just pass the correct one down to the individual line edit forms (along with an "update" method)

Reply:
Nice solution!